### ISSUE TITLE
[Docs] - storybook icons display that no longer exist

### ISSUE THREAD

**Issue by mosheKerbel** [wix/wix-style-react#4207]
Reproduction: go to production storybook and look for Rectangle, these icons does not really exist in wix-ui-icons, we should not display them in storybook.
Notice that it does not reproduce on localhost!
Looks like something is off with the build chain of icons.

Icons section in storybook is showing all the available icons
It does it by importing all icons and iterating over them, [here](https://github.com/wix/wix-style-react/blob/8c5f687d627d81354d33ad9c0207c6cd0876984b/src/new-icons/docs/AllIcons.js#L2)

I thought that a simple change in the import would solve it, but it did not
https://github.com/wix/wix-style-react/pull/4205
You can see in the storybook of this PR that rectangle still exist.

![image](https://user-images.githubusercontent.com/7388217/66267735-5d735780-e83d-11e9-8e87-f70d0fcf80c2.png)

**Comment by argshook** [comment]
tl;dr; `wix-ui-icons-common` exports icons that are proxied by `wix-style-react/new-icons`. Release cycles of those two diverge and boom the issue.

---

this situation can happen, when `wix-ui-icons-common` & `wix-style-react` are released separately.

This is because `wix-style-react` proxies icons with [this](https://github.com/wix/wix-style-react/blob/master/scripts/proxy-wix-ui-icons/index.js) script. that script is run only during `npm run build`.

consider case when `wix-ui-icons-common` has new release but `wix-style-react` doesn't:

* wix-style-react has been released (some time ago) with proxied icons and built storybook (that showcases those icons)
* wix-ui-icons-common has new release with more/less icons
* wix-style-react storybook doesn't care, cause it was built before wix-ui-icons-common was released (and it contains old version of icons in its bundle)

locally it works cause it most likely uses up to date `wix-ui-icons-common` from `node_modules`.

---

possible solutions:

1. release cycles of both libs are somehow done not separately
1. we stop recommending icon usage from `wix-style-react/new-icons` and treat `wix-ui-icons-common` as icons lib which users use separately

**Comment by shlomitc** [comment]
I like the second idea. I would like to create have a codemod not part of WSR.
Can create a new project in `wix-ui` repo dedicated for codemodes like material-ui is doing https://github.com/mui-org/material-ui/tree/master/packages/material-ui-codemod

Also, need to change also our docs and codebase to import from wix-ui-icons common.

Once this storybook feature will be implemented, it will be great as we could implement the docs page from there https://twitter.com/NorbertdeLangen/status/1181295451091410944

I'll prioritize the task

**Comment by simast** [comment]
Currently working on this issue, here is my plan:

For `wix-style-react 7.x`:
- [ ] Update documentation mentioning that icons should be imported from `wix-ui-icons-common` directly and that the old `new-icons/*` method is deprecated.
- [ ] Switch all internal icon imports in `wix-style-react` to use `wix-ui-icons-common` directly.
- [ ] Write a codemod for users to migrate from old `wix-style-react/new-icons/*` to new `wix-ui-icons-common/*` imports.
- [ ] Add a deprecation warning to console if users are including icons from `new-icons/*` proxy?

For `wix-style-react 8.0`:
- [ ] Remove everything related to `new-icons` proxy (backwards-incompatible change).
